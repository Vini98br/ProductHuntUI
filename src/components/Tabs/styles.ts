import styled from "styled-components";

export const Tabs = styled.div`
  width: 100%;
  height: auto;
  background-color: ${(props) => props.theme.colors.secondary};
  display: flex;
`;

export const Tab = styled.div<{ active?: boolean }>`
  color: ${({ active, theme }) =>
    active ? theme.colors.primary : theme.colors.gray_2};
  width: 50%;
  display: flex;
  justify-content: center;
  padding: 15px;
  font-weight: 600;
  cursor: pointer;
  border-bottom: ${(props) =>
    props.active ? `3px solid ${props.theme.colors.primary}` : "unset"};
`;