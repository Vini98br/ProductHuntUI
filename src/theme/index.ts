export const theme = {
  colors: {
    primary: '#D9552E',
    secondary: '#FFFFFF',
    dark_gray: '#1E232A',
    gray: '#76868E',
    gray_2: '#88969D',
    light_gray: '#F4F5F6', 
  }
}

export type AppTheme = typeof theme;