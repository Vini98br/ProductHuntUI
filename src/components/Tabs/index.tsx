import React from 'react';
import { TabEnum } from '../../@types/tab';

import { Tabs as TabsWrapper, Tab } from './styles';

export interface TabsProps {
  tabs: {
    label: string;
    value: TabEnum;
  }[];
  active: TabEnum;
  handleTabClick?: (state: TabEnum) => void;
}

const Tabs: React.FC<TabsProps> = ({ handleTabClick = () => ({}), active, tabs }) => {
  return (
    <TabsWrapper>
      {tabs.map(tab => (
        <Tab
          key={tab.value}
          active={active === tab.value}
          onClick={() => handleTabClick(tab.value)}
          id={`${tab.value}-tab`}
        >
          {tab.label}
        </Tab>
      ))}
    </TabsWrapper>
  );
}

export default Tabs;