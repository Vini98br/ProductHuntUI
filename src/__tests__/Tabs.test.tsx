import React from "react";
import { create, act } from "react-test-renderer";
import { theme } from "../theme";
import { ThemeProvider } from "styled-components";
import Tabs from "../components/Tabs";

test("Render Tabs", async () => {
  const { root } = create(
    <ThemeProvider theme={theme}>
      <Tabs
        active={"popular"}
        tabs={[
          { label: "Popular", value: "popular" },
          { label: "Newest", value: "newest" },
        ]}
      />
    </ThemeProvider>
  );
  const popularTab = root.findByProps({id: "popular-tab"});
  const newestTab = root.findByProps({id: "newest-tab"});

  expect(popularTab.props.children).toEqual("Popular");
  expect(newestTab.props.children).toEqual("Newest");
});

test("Change tabs", async () => {
  let root;
  act(() => {
    root = create(
      <ThemeProvider theme={theme}>
        <Tabs
          active={"popular"}
          tabs={[
            { label: "Popular", value: "popular" },
            { label: "Newest", value: "newest" },
          ]}
        />
      </ThemeProvider>
    );
  });

  // make assertions on root
  expect(root.toJSON()).toMatchSnapshot();

  act(() => {
    root.update(
      <ThemeProvider theme={theme}>
        <Tabs
          active={"newest"}
          tabs={[
            { label: "Popular", value: "popular" },
            { label: "Newest", value: "newest" },
          ]}
        />
      </ThemeProvider>
    );
  });

  // make assertions on root
  expect(root.toJSON()).toMatchSnapshot();
});
