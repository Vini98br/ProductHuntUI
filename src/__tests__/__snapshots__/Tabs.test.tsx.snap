// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Change tabs 1`] = `
<div
  className="sc-bdfBwQ cpBwlf"
>
  <div
    className="sc-gsTCUz beNedl"
    id="popular-tab"
    onClick={[Function]}
  >
    Popular
  </div>
  <div
    className="sc-gsTCUz cIjvaz"
    id="newest-tab"
    onClick={[Function]}
  >
    Newest
  </div>
</div>
`;

exports[`Change tabs 2`] = `
<div
  className="sc-bdfBwQ cpBwlf"
>
  <div
    className="sc-gsTCUz cIjvaz"
    id="popular-tab"
    onClick={[Function]}
  >
    Popular
  </div>
  <div
    className="sc-gsTCUz beNedl"
    id="newest-tab"
    onClick={[Function]}
  >
    Newest
  </div>
</div>
`;
